web/public/docs/api.md

```diff
@@ -241,180 +259,378 @@ multiple times to overwrite the previous version.
 ```javascript
-//Javascript
 windowId = await api.updateWindow({id: windowId, name: 'new window', type: 'Image Window', w:7, h:7, data: {image: ...}})
 ```
+[Try yourself in the createWindow example >>](https://imjoy.io/#/app?plugin=oeway/ImJoy-Demo-Plugins:createWindow&w=examples)
 
-```python
-# Python
 
+Update a window in Python
+```python
 # pass as a dictionary
-windowId = await api.updateWindow({'id': windowId, name: 'new window', type: 'Image Window', w:7, h:7, 'data': {'image': ...}})
+window_id = await api.updateWindow({'id': window_id, name: 'new window', type: 'Image Window', w:7, h:7, 'data': {'image': ...}})
 
 # or named arguments
-windowId = await api.updateWindow(id=windowId, name='new window', type='Image Window', w=7, h=7, data={'image': ...})
+window_id = await api.updateWindow(id=window_id, name='new window', type='Image Window', w=7, h=7, data={'image': ...})
 
 # callback style
 def cb(wid):
-    windowId = wid
-api.updateWindow({'id': windowId, name: 'new window', type: 'Image Window', w:7, h:7, 'data': {'image': ...}}).then(cb)
+    window_id = wid
+api.updateWindow({'id': window_id, name: 'new window', type: 'Image Window', w:7, h:7, 'data': {'image': ...}}).then(cb)
 ```
 
-
-## `api.showDialog(...)`
-show a dialog with customized GUI, example:
+### api.showDialog
+Show a dialog with customized GUI. The answer is stored in the returned object,
+and can be retrieved with the specified `id`. To consider the case when the user
+presses `cancel`, you can use the `try catch` (JavaScript) or `try except` (Python)
+syntax.
 
 ```javascript
 const result = await api.showDialog({
    "name": "This is a dialog",
    "ui": "Hey, please select a value for sigma: {id:'sigma', type:'choose', options:['1', '3'], placeholder: '1'}.",
 })
 ```
-## `api.showProgress(...)`
-update the progress bar on the Imjoy GUI, example: `api.showProgress(85)`
+[Try yourself >>](https://imjoy.io/#/app?plugin=oeway/ImJoy-Demo-Plugins:showDialog&w=examples)
 
-## `api.showStatus(...)`
-update the status text on the Imjoy GUI, example: `api.showStatus('processing...')`
 
-## `api.showSnackbar(...)`
-show a quick message with a snackbar and disappear in a few seconds, example: `api.showSnackbar('processing...', 5)`
+### api.showProgress
+``` javascript
+api.showProgress(progress)
+```
+Updates the progress bar on the Imjoy GUI.
 
-## `api.showPluginProgress(...)`
-update the progress bar of the current plugin (in the plugin menu), example: `api.showPluginProgress(85)`
+**Arguments**
 
-## `api.showPluginStatus(...)`
-update the status text of the current plugin (in the plugin menu), example: `api.showPluginStatus('processing...')`
+* **progress**: Float. Progress in percentage. Allowed range from 0 to 100.
+
+**Examples**
+``` javascript
+api.showProgress(85)
+```
+[Try yourself >>](https://imjoy.io/#/app?plugin=oeway/ImJoy-Demo-Plugins:showProgress&w=examples)
+
+
+### api.showSnackbar
+``` javascript
+api.showSnackbar(message,duration)
```

Comment:
Related to https://github.com/oeway/ImJoy/issues/45 , I think you can start to mention that the unit of duration is seconds for `api.showSnackbar`, I will soon get the the PR merge.